pkg/operator/controller/externaldns/deployment.go

```diff
@@ -265,8 +296,24 @@ func (r *reconciler) updateExternalDNSDeployment(ctx context.Context, current, d
 // Returns a boolean if an update is necessary, and the deployment resource to update to.
 func externalDNSDeploymentChanged(current, expected *appsv1.Deployment) (bool, *appsv1.Deployment) {
 	updated := current.DeepCopy()
+	return externalDNSAnnotationsChanged(current, expected, updated) || externalDNSContainersChanged(current, expected, updated), updated
+}
 
-	return externalDNSContainersChanged(current, expected, updated), updated
+// externalDNSAnnotationsChanged returns true if the current secret annotation differ from the expected
+func externalDNSAnnotationsChanged(current, expected, updated *appsv1.Deployment) bool {
+	changed := false
+	if current.Annotations == nil {
+		updated.Annotations = expected.Annotations
+		return true
+	}
+	for expectedKey, expectedValue := range expected.Annotations {
+		currentVal, currentExists := current.Annotations[expectedKey]
+		if !currentExists || currentVal != expectedValue {
+			updated.Annotations[expectedKey] = expectedValue
+			return true
+		}
+	}
+	return changed
```

Comment:
```suggestion
	return false
```
`changed` variable is used only here, can return `false` directly.